libraries/AP_Logger/AP_Logger_Block.cpp

```diff
@@ -243,6 +248,42 @@ bool AP_Logger_Block::NeedErase(void)
     return true;
 }
 
+/*
+ * iterate through all of the logs files looking for ones that are corrupted and correct.
+ */
+void AP_Logger_Block::validate_log_structure()
+{
+    WITH_SEMAPHORE(sem);
+    bool wrapped = check_wrapped();
+    uint32_t page = 1;
+
+    StartRead(page);
+    uint16_t file = GetFileNumber();
+    uint16_t nextFile = file;
+
+    while (file != 0xFFFF && page <= df_NumPages && (file == nextFile || (wrapped && file < nextFile))) {
+        page = find_last_page_of_log(file);
+        if (page == 0) { // this can happen and may be responsible for corruption that we have seen
+            break;
+        }
+        page++;
+        StartRead(page);
+        file = GetFileNumber();
+        nextFile++;
+        if (wrapped && file < nextFile) {
+            nextFile = file;
+        }
+        if (file == nextFile) {
+            hal.console->printf("Found complete log %d at %X-%X\n", int(file), unsigned(page), unsigned(find_last_page_of_log(file)));
```

Comment:
I think it would be better to use GCS send_text() so these are sent by mavlink (there are several printf like this)

Reply:
Ok, I was concerned about printing a message for every log, I guess I guess summarize the total number of complete logs found.

Reply:
Ok I have converted a few select messages to mavlink messages